src/components/CaretContainer.vue

```diff
@@ -266,26 +267,48 @@ export default Vue.extend({
                 this.doPaste();
             }
         },
-        
+
         doPaste(skipDisableCheck?: boolean) : void {
+            let pasteDestination: CurrentFrame = {id: this.frameId, caretPosition: this.caretAssignedPosition};            
+            const stateBeforeChanges = JSON.parse(JSON.stringify(this.appStore.$state));
```

Comment:
I think we can use lodash's cloneDeep for this but I don't have a strong preference.